docs/src/modules/java-protobuf/pages/actions.adoc

```diff
@@ -360,6 +360,20 @@ action returning and the `addItem` call from the action would lead to more items
 
 Such validation that depends on state can only be done safely while handling the command inside the entity.
 
+== Actions as Life-cycle Hooks
+
+An Action's gRPC method can be triggered automatically when some predefined service life-cycle event happens (currently, only on startup is available), serving as a custom hook. For such use, the method needs be annotated with a specific flag as shown below and its input type must be `google.protobuf.Empty`.
+
+[source,proto]
+----
+include::example$java-protobuf-doc-snippets/src/main/proto/com/example/on_startup.proto[tag=hook]
+----
+<1> Only methods belonging to an Action can be configured as a hook.
+<2> The method must receive `google.protobuf.Empty` as input type.
+<3> This hook will be triggered once the instance startup is concluded (i.e. will be called 3 times if 3 instances are running).
```

Comment:
I think this deserves even more explanation it's not only about the startup but also restarts (sometimes not triggered by the user) and scaling up.

Reply:
yeah, I think you're right, let me give it another try. :)